pinot-controller/src/main/java/org/apache/pinot/controller/helix/core/realtime/SegmentCompletionManager.java

```diff
@@ -1028,7 +1028,7 @@ private int numReplicasToLookFor() {
       _state = State.COMMITTING;
       // In case of splitCommit, the segment is uploaded to a unique file name indicated by segmentLocation,
       // so we need to move the segment file to its permanent location first before committing the metadata.
-      if (isSplitCommit) {
+      if (isSplitCommit && reqParams.getSegmentUploadToController()) {
```

Comment:
In case of split commit, it is the choice of the server as to where it uploads the segment. It could be to a deep store. In which case, you may still need a move from temporary?

Reply:
Only when the server chooses to upload the segment to controller, there is still a need for move from tmp location to the final location. This is why I add an additional checking here. 

If the server uploads the segment to deep store directly, there is no need to move anymore.